api/applicationserver_integrations_storage.proto

```diff
@@ -65,6 +66,9 @@ message GetStoredApplicationUpRequest {
   // The names of the upstream message fields that should be returned. See the API reference
   // for allowed field names for each type of upstream message.
   google.protobuf.FieldMask field_mask = 9 [(gogoproto.nullable) = false];
+
+  // Query upstream messages that have arrived in the last minutes or hours. Cannot be used in conjunction with after and before.
+  google.protobuf.Duration last = 10;
```

Comment:
You can enforce that "either/or" constraint by putting it in a `oneof`.

Reply:
How to keep that compatible? It's a oneof with two fields (`after` and `before`) vs one field (`last`).

Reply:
Good point. I guess it'll just be messy 😞